### PR TITLE
Update shortener

### PR DESCRIPTION
**Pull request by gal**
Implemented delete request, removed domain field in JSON POST request, added basicauth, and added viper vars

### CODE REVIEW

File: commands/shortenCommand.go [hunk 1]
```diff
@@ -48,20 +73,16 @@ func shortenCommand(ctx context.Context, s *discordgo.Session, m *discordgo.Mess
...
-	values := map[string]string{"Slug": slug, "Domain": domain, "Target": originalURL}
+	values := map[string]string{"slug": slug, "url": originalURL}
 	jsonValue, _ := json.Marshal(values)
 	req, err := http.NewRequest("POST", "https://"+viper.GetString("shorten.host"), bytes.NewBuffer(jsonValue))
 	req.Header.Set("Content-Type", "application/json")
 
 	client := &http.Client{}
 	resp, err := client.Do(req)
 	if err != nil {
-		s.ChannelMessageSend(m.ChannelID, "Could not reach URL shortening server.")
```
Comment (oofsauce) [inline commands/shortenCommand.go:62]: is error log here not still needed?

Comment (gal) [reply commands/shortenCommand.go]: yeess

File: commands/shortenCommand.go [hunk 2]
```diff
@@ -33,6 +33,32 @@ func shortenCommand(ctx context.Context, s *discordgo.Session, m *discordgo.Mess
...
+			return
+		}
+		client := http.Client{}
+		resp, err := client.Do(req)
+		if err != nil {
+			s.ChannelMessageSend(m.ChannelID, "Could not reach URL shortening server.")
+			log.WithContext(ctx).WithError(err).Error("Error communicating with shorten server")
+			return
+		}
+		if resp.StatusCode != 202 {
```
Comment (ericm) [inline commands/shortenCommand.go:53]: Use the http lib constant

File: commands/shortenCommand.go [hunk 3]
```diff
@@ -33,6 +33,32 @@ func shortenCommand(ctx context.Context, s *discordgo.Session, m *discordgo.Mess
 	slug (default domain will be used here)
 	*/
 
+	// delete option
+	if params[1] == "delete" {
+		slug := params[2]
```
Comment (ericm) [inline commands/shortenCommand.go:38]: Make sure `len(params) > 2`

File: commands/shortenCommand.go (hunk 1 above)
Comment (oofsauce) [reply commands/shortenCommand.go]: :)

**Review by ericm — CHANGES_REQUESTED** [review]

File: commands/shortenCommand.go [hunk 4]
```diff
@@ -34,7 +34,7 @@ func shortenCommand(ctx context.Context, s *discordgo.Session, m *discordgo.Mess
 	*/
 
 	// delete option
-	if params[1] == "delete" {
+	if len(params) > 1 && params[1] == "delete" {
```
Comment (oofsauce) [inline commands/shortenCommand.go:37]: need to add logic specifically for invalid param length ("no slug specified")

**Review by oofsauce — CHANGES_REQUESTED** [review]

File: commands/shortenCommand.go (hunk 4 above)
Comment (gal) [reply commands/shortenCommand.go]: image

File: commands/shortenCommand.go [hunk 5]
```diff
@@ -21,28 +21,39 @@ func shortenCommand(ctx context.Context, s *discordgo.Session, m *discordgo.Mess
 	}
 
 	params := strings.Split(m.Content, " ")
-	if len(params) == 1 {
-		s.ChannelMessageSend(m.ChannelID, "Missing argument: original-url")
+	if len(params) < 3 {
+		s.ChannelMessageSend(m.ChannelID, "Missing parameter: shortened-url")
```
Comment (oofsauce) [inline commands/shortenCommand.go:25]: error message here doesnt make sense anymore

**Review by oofsauce — CHANGES_REQUESTED** [review]

File: commands/shortenCommand.go (hunk 5 above)
Comment (oofsauce) [reply commands/shortenCommand.go:25]: also surely more detailed syntax errors are nice

Comment (gal) [reply commands/shortenCommand.go:25]: !shorten https://url.com slug
!shorten delete slug

I need to add that to register commands

**Review by oofsauce — COMMENTED** [review]
delete error for slug not found?

File: commands/shortenCommand.go [hunk 6]
```diff
@@ -64,16 +64,18 @@ func shortenCommand(ctx context.Context, s *discordgo.Session, m *discordgo.Mess
...
 		}
-		if resp.StatusCode != http.StatusAccepted {
+		switch resp.StatusCode {
+		case http.StatusAccepted:
+			s.ChannelMessageSend(m.ChannelID, "Deleted shortened URL!")
+			break
+		case http.StatusNotFound:
+			s.ChannelMessageSend(m.ChannelID, "Couldn't find given shortened URL.")
+		default:
 			s.ChannelMessageSend(m.ChannelID, "Error deleting shortened URL.")
```
Comment (oofsauce) [inline commands/shortenCommand.go:74]: put status code in default case just in case

**Review by oofsauce — CHANGES_REQUESTED** [review]

**Review by oofsauce — APPROVED** [review]

**Review by ericm — APPROVED** [review]

**Pull request merged**